src/urllib3/util/connection.py

```diff
@@ -58,6 +61,11 @@ def create_connection(
     # The original create_connection function always returns all records.
     family = allowed_gai_family()
 
+    try:
+        host.encode("idna")
+    except UnicodeError:
+        raise LocationParseError("Host '%s' is not a valid IDNA label" % host)
```

Comment:
1. I think the error message should say "is not a valid IDNA name".
2. Since `LocationParseError` is not a subclass of `UnicodeError` callers that relied on the previous behavior (`except UnicodeError`) will fail to catch the new error now.

Reply:
> Since LocationParseError is not a subclass of UnicodeError callers that relied on the previous behavior (except UnicodeError) will fail to catch the new error now.

Yes, you are right. What should we do? raise `UnicodeError` instead?

Reply:
You can change `LocationParseError` to subclass `UnicodeError` as well.

Regarding my first comment, I see now that when we normalize non-ascii hosts we use the term "label" :raised_eyebrow: :
https://github.com/urllib3/urllib3/blob/6c429006c73da330b3c7e5b349a9a4b0eaf5d609/src/urllib3/util/url.py#L315
so I guess "label" is fine.

Reply:
I made the `LocationParseError` subclass of `UnicodeError`.